rotkehlchen/tests/unit/test_cost_basis.py

```diff
@@ -525,6 +561,120 @@ def test_accounting_lifo_order(accountant):
     ]
 
 
+def test_accounting_hifo_order(accountant):
+    asset = A_BTC
+    cost_basis = accountant.pots[0].cost_basis
+    cost_basis.reset(DBSettings(cost_basis_method=CostBasisMethod.HIFO))
+    asset_events = cost_basis.get_events(asset)
+    # first we do a simple test that from 2 events the one with the highest amount is used
+    event1 = AssetAcquisitionEvent(
+        amount=ONE,
+        timestamp=1,
+        rate=ONE,
+        index=1,
+    )
+    event2 = AssetAcquisitionEvent(
+        amount=FVal('2.0000'),
+        timestamp=2,
+        rate=ONE,
+        index=2,
+    )
+    asset_events.acquisitions_manager.add_acquisition(event1)
+    asset_events.acquisitions_manager.add_acquisition(event2)
+    assert cost_basis.reduce_asset_amount(asset, ONE, 0)
+    acquisitions = asset_events.acquisitions_manager.get_acquisitions()
+    assert len(acquisitions) == 2 and acquisitions[0] == event2 and acquisitions[1] == event1
+
+    # then test to reset
```

Comment:
split this into 2 tests. One `test_simple_hifo_order` and other `test_hifo_order`